### DISCUSSION TITLE
Community contributions and PRs

### DISCUSSION THREAD

**Discussion by nvnivs** [TechToSpeech/terraform-aws-serverless-static-wordpress#49] [Q&A] [answered]
Wasn't to sure if discussions is the right place for this question or issues, but felt better here.

I was wondering, is this project open for contributions from the community in the form of PRs?

Just started having a play with it and already started tweaking some things. Would rather not fork and go off on a tangent so would like to know if that is something of interest here.

Posting the question as there's no `CONTRIBUTING.md` or any PR's merged from other members.

**Comment by petewilcock** [comment, marked as answer]
Absolutely @z0c ! Contributions are very welcome, and the number of forks makes me think I should indeed have a contributions guide. 

I've been working on a couple of features recently, the main one being upgrading to v4 of the AWS Terraform Provider, which will necessitate some state migration for any existing users upgrading - I've got an upgrade guide in progress. I'm also adding support for Graviton-based build agents in CodeBuild and also Graviton Fargate. 

As a general guide pending a proper `CONTRIBUTIONS.md` file, my main asks would be:

* Make sure the feature being developed exists as an issue (or create one if absent), so I can tag it and add it to the [Roadmap](https://github.com/TechToSpeech/terraform-aws-serverless-static-wordpress/projects/1) and reference it in the PR using the [Github guidance](https://github.blog/2013-05-14-closing-issues-via-pull-requests/). 
* Try to make all new functionality optional and toggle-able via a variable flag.
* If adding features which are not present in every region, look to have region-specific logic for alternatives or fallbacks. 

And otherwise all advancement is much appreciated and welcome!

**Comment by nvnivs** [comment]
Awsome!

Makes a lot of sense to first triage any changes through the issues. Going to cross refference the tweaks i currently have and considering with the current issues and roadmap.

**Reply by petewilcock** [reply]
Just to pick up on this, I've started merging features and fixes into a 0.2.0 branch if you would like to contribute, open a PR to this branch:
https://github.com/TechToSpeech/terraform-aws-serverless-static-wordpress/pull/55

**Reply by nvnivs** [reply]
Thanks @petewilcock!

I've started to work on a fix for https://github.com/TechToSpeech/terraform-aws-serverless-static-wordpress/issues/50, will branch of `release/0.2.0` instead.
